processing/src/main/java/org/apache/druid/collections/spatial/ImmutableFloatNode.java

```diff
@@ -182,15 +182,15 @@ public ImmutableNode<float[]> next()
                   numDims,
                   initialOffset,
                   data.getInt(childrenOffset + (count++) * Integer.BYTES),
-                  data,
+                  data.duplicate(),
```

Comment:
it looks like most of the usages of `ByteBuffer` by `ImmutableFloatNode` use calls that specify the offset in the buffer except for `getImmutableBitmap()`, which sets the position and slices it (for some reason, it seems like this method in general could be cleaned up). I wonder if we just change that instead of calling duplicate inside the iterator, so that way it is a bit lazier?

```
  @Override
  public ImmutableBitmap getImmutableBitmap()
  {
    final int sizePosition = initialOffset + offsetFromInitial + HEADER_NUM_BYTES + 2 * numDims * Float.BYTES;
    int numBytes = data.getInt(sizePosition);
    final ByteBuffer readOnlyBuffer = data.asReadOnlyBuffer();
    readOnlyBuffer.position(sizePosition + Integer.BYTES);
    readOnlyBuffer.limit(readOnlyBuffer.position() + numBytes);
    return bitmapFactory.mapImmutableBitmap(readOnlyBuffer);
  }
```

Reply:
Though the current way, also avoids any potential misuse of the buffer in this class.

Reply:
yea, but it would call duplicate for every entry instead of just the ones that satisfy the bounds check that we need a bitmap for

Reply:
@clintropolis  Thank you Clint, I liked the suggestion as well as it can avoid creating duplicate references on each tree node and sometimes tree can be really big.